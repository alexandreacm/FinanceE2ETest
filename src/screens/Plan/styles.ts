import { StyleSheet } from 'react-native';
import { colors } from '../../theme';

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 24,
    backgroundColor: colors.WHITE
  },
  options: {
    width: 200,
    backgroundColor: colors.OPTIONS,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingHorizontal: 15,
    paddingVertical: 7,
    borderRadius: 5,
    alignSelf: 'flex-end',
    marginTop: 25
  },
  details: {
    color: colors.TEXT_NOTE,
    textAlign: 'center',
    fontSize: 12,
    marginTop: 22
  },
  confirmation: {
    fontWeight: 'bold',
    fontSize: 16,
    textAlign: 'center',
    marginTop: 12,
    color: colors.BUTTON,
    marginBottom: 12
  }
});