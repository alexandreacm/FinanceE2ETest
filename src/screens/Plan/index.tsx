import React, { useState } from 'react';
import { View, Text, TouchableWithoutFeedback, Keyboard } from 'react-native';

import { Input } from '../../components/Input';
import { Header } from '../../components/Header';
import { Option } from '../../components/Option';
import { Button } from '../../components/Button';
import { PlanInfo, PlanInfoProps } from '../../components/PlanInfo';

import { styles } from './styles';

export default function Plan() {
  const [email, setEmail] = useState<string>('');
  const [plan, setPlan] = useState<PlanInfoProps>({ name: 'Basic', value: '5.25' });
  const [emailSent, setEmailSent] = useState(false);
  const [validateEmail, setValidateEmail] = useState(false);

  function handleChangePlan(plan: string) {
    setPlan({
      name: plan === 'basic' ? 'Basic' : 'Premium',
      value: plan === 'basic' ? '5.25' : '6.99'
    });
  }

  function handleSubscribe() {
    if(email === ''){
       setValidateEmail(true);
       setEmailSent(false);
       return
    }
    setValidateEmail(false)
    setEmailSent(true);
  }

  return (
    <TouchableWithoutFeedback testID="TouchableWithoutKeyboard" onPress={Keyboard.dismiss}>
      <View style={styles.container}>
        <Header />

        <PlanInfo
          name={plan.name}
          value={plan.value}
        />

        <View style={styles.options}>
          <Option
            testID="option-premium"
            title="Premium"
            active={plan.name === 'Premium'}
            onPress={() => handleChangePlan('premium')}
          />
          <Option
            testID="option-basic"
            title="Basic"
            active={plan.name === 'Basic'}
            onPress={() => handleChangePlan('basic')}
          />
        </View>

        <Input
          testID="input-email"
          placeholder="your email"
          required = {validateEmail}
          value={email}
          onChangeText={setEmail}
        />

        {
          emailSent &&
          <Text style={styles.confirmation} testID="confirmation-message">
            We send you a  {'\n'}
            confirmation email.
          </Text>
        }

        <Button
          testID="button-subscribe"
          title="Subscribe"
          onPress={handleSubscribe}
        />

        <Text style={styles.details} testID="plan-note">
          If the price changes, we'll notify you beforehand.
          You cam check your renewal date or cancel anytime via your Account page.
        </Text>
      </View>
    </TouchableWithoutFeedback>
  );
}