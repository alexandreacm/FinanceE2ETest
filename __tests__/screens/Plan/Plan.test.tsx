import * as React from "react";
import { create } from "react-test-renderer";
import { render, fireEvent } from "@testing-library/react-native";

import Plan from "../../../App";

describe("renders correctly", () => {
  const tree = create(<Plan />).toJSON();

  it("Should renders correctly app component", () => {
    expect(tree).toMatchSnapshot();
  });
});
