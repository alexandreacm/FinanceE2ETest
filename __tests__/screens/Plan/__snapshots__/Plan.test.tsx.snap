// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Should renders correctly app component 1`] = `
Object {
  "name": "Alexandre",
}
`;
